### PR TITLE
Get the maximum FizzBuzz from console input

### PR DESCRIPTION
**Pull request by benjimarshall**

### CODE REVIEW

File: FizzBuzz/Program.cs [hunk 1]
```diff
@@ -8,12 +8,37 @@ class Program
...
             }
         }
 
+        static int GetMaxFromUser()
+        {
+            while (true)
+            {
+                Console.Write("Maximum FizzBuzz number: ");
+                var inputLine = Console.ReadLine();
+                if (int.TryParse(inputLine, out var inputInt))
```
Comment (FridaTveit) [inline FizzBuzz/Program.cs:24]: I would probably use `int.Parse` with a try/catch. Having `out` variables is slightly less used and therefore makes the code a bit less readable I think 🙂

File: FizzBuzz/Program.cs [hunk 2]
```diff
@@ -8,12 +8,37 @@ class Program
...
+                {
+                    if (inputInt >= 1)
```
Comment (FridaTveit) [inline FizzBuzz/Program.cs:26]: There are quite a few levels of nesting here, which generally makes the code harder to read and understand. Can you rewrite it to have less nesting? 🙂

**Review by FridaTveit — COMMENTED** [review]
A few suggestions, otherwise looks good 🙂

File: FizzBuzz/Program.cs (hunk 1 above)
Comment (benjimarshall) [reply FizzBuzz/Program.cs]: Fair enough, I've just changed that in edaf668

File: FizzBuzz/Program.cs (hunk 2 above)
Comment (benjimarshall) [reply FizzBuzz/Program.cs]: I've changed this bit of code; it's at the same level of nesting but one of the levels is a straightforward try block. Let me know if you want me to have another go 🙂

Comment (FridaTveit) [reply FizzBuzz/Program.cs]: You're right it is quite straight forward, however I would still prefer for it to have less nesting. I think maybe separating the parsing of the input from printing error/returning results would help? E.g. (in pseudocode):
```
var input = ParseInput();
if (input invalid int)
{
  print error;
}
else if (input 0 or negative)
{
  print error;
}
else 
{
  return input;
}
```
You'd still have two levels of nesting, but that's better than three 🙂  What do you think?

File: FizzBuzz/Program.cs [hunk 3]
```diff
@@ -8,12 +8,36 @@ class Program
...
             {
                 Console.WriteLine(FizzBuzz(i));
             }
         }
 
+        static int GetMaxFromUser()
+        {
+            while (true)
+            {
+                Console.Write("Maximum FizzBuzz number: ");
```
[the remark this answers is not in the document]
Comment (benjimarshall) [reply FizzBuzz/Program.cs:22]: Nullable types are a good way of solving that, I didn't realise they existed in C#. I've done all of that in edb5f5a

File: FizzBuzz/Program.cs [hunk 4]
```diff
@@ -20,21 +20,32 @@ static int GetMaxFromUser()
...
                 {
-                    inputInt = int.Parse(inputLine);
+                    Console.WriteLine("Please enter an integer.");
                 }
-                catch (FormatException e)
+                else if (inputInt >= 1)
                 {
-                    Console.WriteLine("Please enter an integer.");
-                    continue;
+                    return (int)inputInt;
```
Comment (FridaTveit) [inline FizzBuzz/Program.cs:31]: I think you can use `.Value` here to get the `int`? 🙂

**Review by FridaTveit — COMMENTED** [review]
Looks great! 😄  I think there's one tiny improvement you can do, otherwise I think it's ready to merge 🙂

File: FizzBuzz/Program.cs (hunk 4 above)
Comment (benjimarshall) [reply FizzBuzz/Program.cs]: That's a better solution! Fixed in d600a20

**Review by FridaTveit — APPROVED** [review]
Looks great! 🙂

**Pull request merged**